Build currently fails with JDK 8

I just clone the latest `master` but cannot get it to build with a JDK 8 installation:
```shell
$ git log -1 --format='%H' | cat
01a7da249bb3e348565410e99c269006dfdec433

$ mvn --version
Apache Maven 3.9.4 (dfbb324ad4a7c8fb0bf182e6d91b0ae20e3d2dd9)
Maven home: /opt/homebrew/Cellar/maven/3.9.4/libexec
Java version: 1.8.0_345, vendor: Azul Systems, Inc., runtime: /Users/fbiville/.sdkman/candidates/java/8.0.345-zulu/zulu-8.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "13.5.2", arch: "aarch64", family: "mac"

$ mvn verify
[...]
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/fbiville/workspace/json-schema-validator/src/test/java/com/networknt/schema/HTTPServiceSupport.java:[19,19] cannot access io.undertow.Undertow
  bad class file: /Users/fbiville/.m2/repository/io/undertow/undertow-core/2.3.5.Final/undertow-core-2.3.5.Final.jar(io/undertow/Undertow.class)
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```

As hinted by the error message, the build works fine with a JDK 11.

This contradicts the `README`:
> The library supports Java 8 and up. If you want to build from the source code, you need to install JDK 8 locally.

--- a/pom.xml
+++ b/pom.xml
@@ -79,7 +79,7 @@
     <version.logback>1.2.11</version.logback>
     <version.slf4j>2.0.7</version.slf4j>
     <version.surefire>3.0.0</version.surefire>
-    <version.undertow>2.3.5.Final</version.undertow>
+    <version.undertow>2.2.25.Final</version.undertow>
   </properties>
 
   <dependencies>

